### PR TITLE
Fix: one-var allows uninitialized vars in ForIn/ForOf (fixes #5744)

### LINKED ISSUE #5744: one-var reports error for a variable declared in a for-loop after an uninitialized variable declaration
**What version of ESLint are you using?**
v2.5.3

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```json
{
  "rules": {
    "one-var": ["error", {
      "initialized": "never",
      "uninitialized": "always"
    }]
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

Here is a small code snippet reproducing my issue. Just save it to a file and run `eslint` with the config shown above.

```js
function reproduce(candidates) {
  var err;
  for (var ix in candidates) {
    err = ix > 0;
  }
  console.log(err);
}
```

**What did you expect to happen?**

The code should pass linter checks.

**What actually happened? Please include the actual, raw output from ESLint.**

```
$ eslint bug.js

/private/tmp/bug/bug.js
  3:8  error  Combine this with the previous 'var' statement with uninitialized variables  one-var

✖ 1 problem (1 error, 0 warnings)
```

As far as my understanding goes, eslint considers the `ix` variable declared inside `for` loop as unitialized and therefore complains that it should have been defined on the same line as the `err` variable (before the `for` loop). Indeed, the following code passes the checks:

```js
function reproduce(candidates) {
  var err, ix;
  for (ix in candidates) {
    err = ix > 0;
  }
  console.log(err);
}
```

From my point of view, I am proposing one of the following two changes:
 - Fix the implementation of `one-var` to treat `for` loop variables differently.
 - Add an `one-var` option similar to `init-declaration'`s `ignoreForLoopInit` that will tell `one-var` to treat variables defined in for-loops differently.

Thoughts?

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: tests/lib/rules/one-var.js [hunk 1]
```diff
@@ -400,14 +450,6 @@ ruleTester.run("one-var", rule, {
         },
         {
             code: "var i = [0], j;",
-            options: [ { initialized: "never" } ],
```
Comment (kaicataldo) [inline tests/lib/rules/one-var.js]: Looks like this is a duplicate test

Comment (kaicataldo) [inline tests/lib/rules/one-var.js]: Ah, never mind

**Comment by nzakas** [comment]
Code looks good. Can you add something to the docs?

**Comment by kaicataldo** [comment]
Oh yeah - sure thing!

**Comment by kaicataldo** [comment]
Updated with changes to the docs

File: docs/rules/one-var.md [hunk 2]
```diff
@@ -203,6 +203,8 @@ function foo() {
 
 When configured with an object as the first option, you can individually control how `var`, `let`, and `const` are handled, or alternatively how `uninitialized` and `initialized` variables are handled (which if used will override `var`, `let`, and `const`).
 
+**Note:** A variable declared in a for-in or for-of loop will NOT be flagged with the option `{ uninitialized: "always" }`, as this value is determined by the loop.
```
Comment (nzakas) [inline docs/rules/one-var.md]: Can you change "NOT" to lowercase (we don't do this anywhere else)?

Comment (kaicataldo) [inline docs/rules/one-var.md]: Oh yeah, no problem

Comment (kaicataldo) [inline docs/rules/one-var.md]: Fixed

**Comment by nzakas** [comment]
Lgtm

**Pull request merged**